### PR TITLE
Check gtClassProfileCandidateInfo before instrumentation

### PR DESCRIPTION
**Pull request by EgorBo**
`BBF_HAS_CLASS_PROFILE` doesn't mean all virtual and cast calls in the current block are instrumentable.

It fixes the crash I observed in `dotnet-trace`

### CODE REVIEW

**Comment by EgorBo** [comment]
cc @AndyAyersMS @jakobbotsch

File: src/coreclr/jit/fgprofile.cpp [hunk 1]
```diff
@@ -1448,9 +1448,10 @@ class ClassProbeVisitor final : public GenTreeVisitor<ClassProbeVisitor<TFunctor
     Compiler::fgWalkResult PreOrderVisit(GenTree** use, GenTree* user)
     {
         GenTree* const node = *use;
-        if (node->IsCall())
+        if (node->IsCall() && (node->AsCall()->gtCallMoreFlags & GTF_CALL_M_CLASS_PROBE))
```
Comment (jakobbotsch) [inline src/coreclr/jit/fgprofile.cpp:1451]: Can we just check `call->gtClassProfileCandidateInfo != nullptr` instead of adding another flag?

Comment (AndyAyersMS) [reply src/coreclr/jit/fgprofile.cpp:1451]: That sounds reasonable to me too.

Comment (EgorBo) [reply src/coreclr/jit/fgprofile.cpp]: @jakobbotsch @AndyAyersMS done

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**